### PR TITLE
[SDK] Improve IsReactOS function

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Implement a accurate version check helper.
JIRA issue: N/A

## Proposed changes

- Use `shlwapi!SHGetValueW` to check the OS.

## TODO

- [ ] Do build.

### CODE REVIEW

**Review by HBelusca — CHANGES_REQUESTED** [review]
This won't work because this relies on an optional binary we include only if the rosapps subdir is compiled (which it is not by default, e.g. in the test builds).

**Comment by whindsaks** [comment]
For the moment, checking if system32\kernel32_vista.dll exists on disk is probably enough?

Checking the company name in the version info of kernel32.dll is probably better. If that is too slow, an exported custom function could be added to kernel32.dll? 

Wine has `wine_get_version` in ntdll...

**Comment by HBelusca** [comment]
> For the moment, checking if system32\kernel32_vista.dll exists on disk is probably enough?
> 
> Checking the company name in the version info of kernel32.dll is probably better. If that is too slow, an exported custom function could be added to kernel32.dll?
> 
> Wine has `wine_get_version` in ntdll...

Please see the review comments in this PR first: https://github.com/reactos/reactos/pull/4411

**Comment by katahiromz** [comment]
Used `SHGetValueW`.

File: sdk/include/psdk/versionhelpers.h [hunk 1]
```diff
@@ -145,12 +145,32 @@ IsActiveSessionCountLimited()
...
 #ifdef __REACTOS__
-VERSIONHELPERAPI
-IsReactOS()
-{
-    // FIXME: Find a better method!
-    WCHAR szWinDir[MAX_PATH];
-    GetWindowsDirectoryW(szWinDir, _countof(szWinDir));
-    return (wcsstr(szWinDir, L"ReactOS") != NULL);
+#include <winreg.h>
+typedef DWORD (WINAPI FN_SHGetValueW(HKEY hKey, LPCWSTR lpszSubKey, LPCWSTR lpszValue,
```
Comment (whindsaks) [inline sdk/include/psdk/versionhelpers.h:149]: This "leaks" the function type to global scope. Declare it inside the function...

**Pull request closed without merging**